src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/DeprecationNotice.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bridge\PhpUnit\DeprecationErrorHandler;
+
+final class DeprecationNotice
+{
+    /**
+     * @var int
+     */
+    private $count = 0;
+
+    /**
+     * @var array<string,int>
+     */
+    private $countsByCaller = [];
+
+    public function addObjectOccurence($class, $method)
+    {
+        if (!isset($this->countsByCaller["$class::$method"])) {
+            $this->countsByCaller["$class::$method"] = 0;
+        }
+        ++$this->countsByCaller["$class::$method"];
+        ++$this->count;
```

Comment:
Naming is confusing to me, why is same variable used for carrying proceduralOccurences too? It can't be procedural occurence if it's incremented on object occurence too, can it? Perhaps it should be just called `totalCount` and `addProceduralOccurence` renamed to `addOccurence`

Reply:
It's `DeprecationNotice::$count`, and can be incremented both because of calls in procedural code or calls in object-oriented  code… no asymmetry here, this looks good to me but maybe I should rename `addObjectOccurence` to `addObjectOrientedOccurence`. Longer but maybe less confusing?